src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultDocumentContextFactory.cs

```diff
@@ -100,12 +101,24 @@ private bool TryResolveDocument(string filePath, VSProjectContext? projectContex
         }
 
         var project = _projectSnapshotManagerAccessor.Instance.GetLoadedProject(projectContext.ToProjectKey());
-        if (project.GetDocument(filePath) is { } document)
+        if (project?.GetDocument(filePath) is { } document)
```

Comment:
`DefaultProjectSnapshotManager` was null enabled and annotated correctly, but the base class was not, so this got missed. It hit for documents where VS knew about the project, but the language server didn't, and brought down the whole server, making CLaSP reboot. This of course meant that the next few requests were to a language server that had no project knowledge, but VS did, so it hit here again, etc. etc.

Reply:
FWIW, I've got a change on my machine that makes this method always non-nullable, which should clean up some of the `?` you need to pepper around.